src/mongo_client.ts

```diff
@@ -448,11 +448,10 @@ export class MongoClient extends TypedEventEmitter<MongoClientEvents> {
       }
     }
 
-    const topology = new Topology(options.hosts, options);
+    const topology = new Topology(this, options.hosts, options);
     // Events can be emitted before initialization is complete so we have to
     // save the reference to the topology on the client ASAP if the event handlers need to access it
     this.topology = topology;
```

Comment:
```suggestion
    this.topology = new Topology(this, options.hosts, options);
```

And update the following code, we don't need a comment that refers to this as a special case anymore I think. Or maybe keep the comment but still immediately assign to the MongoClient.

Reply:
@W-A-James bump, the comment went away but not the code structure: `this.topology = new Topology`